packages/material-ui/src/Button/Button.test.js

```diff
@@ -367,4 +370,121 @@ describe('<Button />', () => {
       expect(markup.text()).to.equal('Hello World');
     });
   });
+
+  describe('custom theme variants', () => {
+    const WrappedComponent = ({ theme, ...props }) => {
+      return (
+        <ThemeProvider theme={theme}>
+          <Button data-testid="component" {...props}>
+            Content
+          </Button>
+        </ThemeProvider>
+      );
+    };
+
+    WrappedComponent.propTypes = {
+      theme: PropTypes.object,
+    };
+
+    it('should map the variant classkey to the component', function test() {
+      if (/jsdom/.test(window.navigator.userAgent)) {
+        // see https://github.com/jsdom/jsdom/issues/2953
+        this.skip();
+      }
+
+      const theme = createMuiTheme({
+        variants: {
+          MuiButton: [
+            {
+              props: { variant: 'test' },
+              styles: { backgroundColor: 'rgb(255, 0, 0)' },
+            },
+          ],
+        },
+      });
+
+      render(<WrappedComponent theme={theme} variant="test" />);
+
+      const style = window.getComputedStyle(screen.getByTestId('component'));
+      expect(style.getPropertyValue('background-color')).to.equal('rgb(255, 0, 0)');
+    });
+
+    it('should map the latest props combination classkey to the component', function test() {
+      if (/jsdom/.test(window.navigator.userAgent)) {
+        // see https://github.com/jsdom/jsdom/issues/2953
+        this.skip();
+      }
+
+      const theme = createMuiTheme({
+        variants: {
+          MuiButton: [
+            {
+              props: { variant: 'test' },
+              styles: { backgroundColor: 'rgb(255, 0, 0)' },
+            },
+            {
+              props: { variant: 'test', size: 'large', color: 'primary' },
```

Comment:
Added test case for a props combination classKey that is not generated by default.